config/url_config_test.go

```diff
@@ -8,17 +8,18 @@ import (
 
 func TestURLConfig(t *testing.T) {
 	u := NewURLConfig(EnvironmentOf(MapFetcher(map[string][]string{
-		"http.key":                           []string{"root", "root-2"},
-		"http.https://host.com.key":          []string{"host", "host-2"},
-		"http.https://user@host.com/a.key":   []string{"user-a", "user-b"},
-		"http.https://user@host.com.key":     []string{"user", "user-2"},
-		"http.https://host.com/a.key":        []string{"host-a", "host-b"},
-		"http.https://host.com:8080.key":     []string{"port", "port-2"},
-		"http.https://host.com/repo.git.key": []string{".git"},
-		"http.https://host.com/repo.key":     []string{"no .git"},
-		"http.https://host.com/repo2.key":    []string{"no .git"},
-		"http.http://host.com/repo.key":      []string{"http"},
-		"http.https://host.*/a.key":          []string{"wild"},
+		"http.key":                                []string{"root", "root-2"},
+		"http.https://host.com.key":               []string{"host", "host-2"},
+		"http.https://user@host.com/a.key":        []string{"user-a", "user-b"},
+		"http.https://user@host.com.key":          []string{"user", "user-2"},
+		"http.https://host.com/a.key":             []string{"host-a", "host-b"},
+		"http.https://host.com:8080.key":          []string{"port", "port-2"},
+		"http.https://host.com/repo.git.key":      []string{".git"},
+		"http.https://host.com/repo.key":          []string{"no .git"},
+		"http.https://host.com/repo2.key":         []string{"no .git"},
+		"http.http://host.com/repo.key":           []string{"http"},
+		"http.https://host.com:443/repo3.git.key": []string{"port"},
+		"http.https://host.*/a.key":               []string{"wild"},
```

Comment:
Since the new `portForURL()` function checks for the URLs with the SSH protocol, might it be advisable to add test or two here which use `ssh://...` format URLs?

Reply:
I included it only for completeness, since I'm pretty sure we only use this code path for `http.*` keys, which don't work for SSH requests.  But I'll add some tests here nevertheless, in case we use it in a situation like the credential codepath.